### PR TITLE
Upgrade to yarn V3 

### LINKED ISSUE #1588: Upgrade to yarn v3
### Please describe the task that needs to be done

currently using yarn v1
migration guide https://yarnpkg.com/getting-started/migration

it will be nice to upgrade other repositories using yarn too.

Advantages:
* Be on active development version. y[arn 1.x is in maintenance mode since January 2020](https://classic.yarnpkg.com/lang/en/docs/install/#debian-stable)

### PR DESCRIPTION
**Pull request by MohammadiIram**
Which issue(s) this PR fixes #1588

### CODE REVIEW

File: .yarnrc.yml [hunk 1]
```diff
@@ -0,0 +1,2 @@
+yarnPath: .yarn/releases/yarn-3.6.0.cjs
```
[the remark this answers is not in the document]
Comment (MohammadiIram) [reply .yarnrc.yml:1]: It is automatically added when I do the migration steps, As stated in the migration procedures, I have just added nodeLinker: node-modules.

[the remark this answers is not in the document]
Comment (igarashitm) [reply .yarnrc.yml:1]: If this file `.yarn/releases/yarn-3.6.0.cjs` could be downloaded as a part of build process, then we'd need to do so. If that's not possible, we'd need to add it to the repo. @lordrip do you know what's the best practice?

File: .gitignore [hunk 2]
```diff
@@ -4,7 +4,7 @@
 /node_modules
 dist
 # no longer using yarn berry
-/.yarn
+.yarn/*
```
[the remark this answers is not in the document]
Comment (MohammadiIram) [reply .gitignore:7]: Sure, I'll do that.

**Comment by MohammadiIram** [comment]
> @MohammadiIram I checked where the yarn cache problem came and I summarized the findings in the following pull request: #1926
> 
> Thanks for the help with this issue, unfortunately, this spiraled into a few hidden challenges that weren't considered at the beginning while writing this task.

Okay, no problem. If you need any help in the future, I'll be happy to help you.

**Comment by lordrip** [comment]
> > @MohammadiIram I checked where the yarn cache problem came and I summarized the findings in the following pull request: #1926
> > Thanks for the help with this issue, unfortunately, this spiraled into a few hidden challenges that weren't considered at the beginning while writing this task.
> 
> Okay, no problem. If you need any help in the future, I'll be happy to help you.

On the contrary, thanks to you @MohammadiIram for taking care of this :bow:, I'm sorry that this situation had hidden challenges :face_exhaling: 

If you want to work on something else, we have a few issues that might be of interest. That being said, feel free to check any issue that sounds interesting to you, even if the topic is not clear enough, we would be happy to improve the definition.

Since I'm not sure about which area will be more fun for you, I'll just drop some ideas and please feel free to check them out at your own pace or request clarifications if needed:

* https://github.com/KaotoIO/kaoto-ui/pull/1926 Checking this might be interesting for seeing where are the challenges that I mentioned before.
* [Good first issues](https://github.com/KaotoIO/kaoto-ui/issues?q=is%3Aopen+is%3Aissue+label%3A%22good+first+issue%22): Here we have what are supposed to be small issues but with good value.
    * https://github.com/KaotoIO/kaoto-ui/issues/1484 could be interesting if you wanna do something around the UI - config area
    * https://github.com/KaotoIO/kaoto-ui/issues/1232 is more focused on Typescript and might be a little bit bigger in scope :smiley: 
* https://github.com/KaotoIO/kaoto-ui/issues/1377 is focused on enabling the linter to check on CI, that said, if there's a pull request that has a different style it should notify it and stop the build.
* https://github.com/KaotoIO/kaoto-ui/issues/1429 is focused on providing a focus trap to the mini-catalog, so the user can navigate using the keyboard arrows

**Comment by lordrip** [comment]
Closing it in favor https://github.com/KaotoIO/kaoto-ui/pull/1926. Once again thanks @MohammadiIram for the valuable help :muscle:

**Pull request closed without merging**